adapters/auth/index.ts

```diff
@@ -0,0 +1,42 @@
+import axios, { isAxiosError } from 'axios'
+import { transformSelfInfoFromApi } from './auth.transformer'
+import { type ApiResponse } from '~/interfaces/ApiResponse'
+import { type User } from '~/interfaces/User'
+import { type SelfInfo } from './auth.type'
+import {
+  type ErrorApiForbidden,
+  type ErrorApiUnauthorized,
+  type ErrorApiUnavailable,
+  type ErrorApiNotFound,
+  type ErrorApiBase
+} from '~/interfaces/ErrorApi'
+
+type GetSelfResponse = ApiResponse<
+{
+  user: User
+  info: SelfInfo
+},
+ErrorApiForbidden | ErrorApiUnauthorized | ErrorApiUnavailable | ErrorApiNotFound | ErrorApiBase
```

Comment:
how does it decide which error response to send?

Reply:
It doesn't. Have to decide how this flow should actually work, but for now this should be a decent starting point.